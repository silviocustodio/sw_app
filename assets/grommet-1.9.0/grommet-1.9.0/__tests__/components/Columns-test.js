// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react-test-renderer';

import Columns from '../../src/js/components/Columns';

// needed because this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('Columns', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <Columns>
        <span>test1</span>
        <span>test2</span>
      </Columns>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correct className', () => {
    const component = renderer.create(
      <Columns className="test">
        <span>test1</span>
        <span>test2</span>
      </Columns>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders microdata properties', () => {
    const component = renderer.create(
      <Columns itemScope={true} itemType="http://schema.org/Article"
        itemProp="test">
        <span>test1</span>
        <span>test2</span>
      </Columns>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders masonry properties', () => {
    const component = renderer.create(
      <Columns masonry={true}>
        <span>test1</span>
        <span>test2</span>
      </Columns>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
