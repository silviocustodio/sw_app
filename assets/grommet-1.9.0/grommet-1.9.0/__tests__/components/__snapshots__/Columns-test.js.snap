// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Columns has correct default options 1`] = `
<div
  className="grommetux-columns grommetux-columns--justify-start grommetux-columns--responsive"
>
  <div
    className="grommetux-columns__column"
  >
    <span>
      test1
    </span>
    <span>
      test2
    </span>
  </div>
</div>
`;

exports[`Columns renders correct className 1`] = `
<div
  className="grommetux-columns grommetux-columns--justify-start grommetux-columns--responsive test"
>
  <div
    className="grommetux-columns__column"
  >
    <span>
      test1
    </span>
    <span>
      test2
    </span>
  </div>
</div>
`;

exports[`Columns renders masonry properties 1`] = `
<div
  className="grommetux-columns grommetux-columns--justify-start grommetux-columns--responsive"
>
  <div
    className="grommetux-columns__column"
  >
    <span>
      test1
    </span>
    <span>
      test2
    </span>
  </div>
</div>
`;

exports[`Columns renders microdata properties 1`] = `
<div
  className="grommetux-columns grommetux-columns--justify-start grommetux-columns--responsive"
  itemProp="test"
  itemScope={true}
  itemType="http://schema.org/Article"
>
  <div
    className="grommetux-columns__column"
  >
    <span>
      test1
    </span>
    <span>
      test2
    </span>
  </div>
</div>
`;
